### DISCUSSION TITLE
Welcome to OnvifDeviceManager Discussions!

### DISCUSSION THREAD

**Discussion by Quedale** [Quedale/OnvifDeviceManager#21] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

If you want to discuss new features or looking for assistance, this is a good place to start before creating an issue.

**Comment by tbfraga** [comment]
please, include "sudo apt install git" into [Mandatory]  package dependencies ... Thanks.

**Reply by Quedale** [reply]
I always took it for granted given that people are reading it from a github repo.
That being said, if I ever create a tar file, the autogen.sh script still needs git to run properly, so I added it.

Thanks  for the feedback!

**Reply by tbfraga** [reply]
I use elementary OS which is based on ubunto. I needed to install all the optional packages to be able to finish the installation. 

But still, the software did not identify anything automatically.

Can you please let me know if there is any manual for ubunto. Or even if I need to change some configuration file.  

I was also wondering if there is a command to automatically uninstall all installed packages. In case I give up the installation.

Lastly, I was wondering if somehow the installed packages leave my computer vulnerable. Elementary OS is very safe, but security is compromised when we install different packages. 

Could you please help me to clear these doubts.

**Reply by tbfraga** [reply]
There is an unofficial version of github desktop for linux. I believe this software dispenses installing the git library. I have been using this tool for a long time. It's really quite useful.

**Comment by jonkirton55** [comment]
First attempt failed install needed "apt-get install libxml2-dev" (don't know if that was right thing to do but it worked!)  Second attempt after deleting the folder worked and is excellent thank you.  Don't really need the login security but that's fine.  Only thing I've found that I can get my camera to work on Mint after recent updates stopped the package I was using.  Thanks!

**Reply by Quedale** [reply]
Sorry for the slow response, I have a busy personal life and I have little time for personal projects that I enjoy.

Libxml2 is most likely required by at least one dependency. I just never noticed it as a requirement most likely because all the distro I tested shipped with it. I'll make the proper adjustments for Mint support when I get a chance.

It is very likely that upgrading system packages/kernel will break a native build. (On some cases)
I would need to maintain multiple distribution packages for each popular distro and I can't commit to this much support by myself unfortunately.
This is why I recommend the flatpak approach for non-developers. I also normally prefer native integration when I can, but flatpak offers a very simple approach to resolving this issue.

I still want to officially publish this app on flathub, but I need to finish a few things before it can be done like creating a webpage.

Regarding the initial password, this key is used to encrypt the data file that contains all your camera passwords. I still want to add an option to disable it for users that dont feel they need it or even use native distro key management features. (You arent the first mentioning this, so I may get to it faster rather than later.

Cheers!
